Add --no-source option flag to rustdoc (#75522)
rust-lang/rust: src/doc/rustdoc/src/command-line-arguments.md

When `rustdoc` receives this flag, it will print an extra "Version (version)" into the sidebar of
the crate root's docs. You can use this flag to differentiate between different versions of your
library's documentation.

## `--no-source`: prevent source code generation in HTML output format generation

> Using this flag looks this this:

Reviewer: Typo:

```suggestion
Using this flag looks like this:
```

```bash
$ rustdoc src/lib.rs --no-source
```

> By default, `rustdoc` generates the source code in the HTML output format. When this flag is

Reviewer 2: This wording is a little odd, it makes it sound like rustdoc is writing code 😆 
```suggestion
By default, `rustdoc` renders the source code as HTML and links to it from the documentation. When this flag is
```

> this flag is used, `rustdoc` won't generate it. `--no-source` is equivalent to

Reviewer: nit: Rewording it this way seems a bit clearer:

```suggestion
this flag is used, `rustdoc` won't render the source code. `--no-source` is equivalent to the
```

> used, `rustdoc` won't generate it. It is an equivalent of the `#![doc(html_no_source)]`

Reviewer 2: ```suggestion
used, `rustdoc` won't generate it. `--no-source` is equivalent to `#![doc(html_no_source)]`
```
